src/kakarot/accounts/account_contract.cairo

```diff
@@ -88,6 +97,69 @@ func is_initialized{
 
 // EOA specific entrypoints
 
+@external
+func execute_from_outside{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, bitwise_ptr: BitwiseBuiltin*, range_check_ptr
+}(
+    outside_execution: OutsideExecution,
+    call_array_len: felt,
+    call_array: CallArray*,
+    calldata_len: felt,
+    calldata: felt*,
+    signature_len: felt,
+    signature: felt*,
+) -> (response_len: felt, response: felt*) {
+    alloc_locals;
+    let (caller) = get_caller_address();
+
+    if (outside_execution.caller != 'ANY_CALLER') {
+        assert caller = outside_execution.caller;
+    }
+
+    let (block_timestamp) = get_block_timestamp();
+    let too_early = is_le(block_timestamp, outside_execution.execute_after);
+    with_attr error_message("Execute from outside: too early") {
+        assert too_early = FALSE;
+    }
+    let too_late = is_le(outside_execution.execute_before, block_timestamp);
+    with_attr error_message("Execute from outside: too late") {
+        assert too_late = FALSE;
+    }
+    with_attr error_message("Execute from outside: multicall not supported") {
+        assert call_array_len = 1;
+    }
+
```

Comment:
Here maybe I missed something but shouldn't we maintain something to check that the outside nonce doesn't exist in like our base a little bit like this is done in the spec with argent:

```
let nonce = outside_execution.nonce;
    assert(!self.outside_nonces.read(nonce), 'argent/duplicated-outside-nonce');
```

Reply:
Yes indeed, according to the SNIP9, the outside nonce must be unique.
The nonce will be checked against the tx signer nonce (EOA ethereum) in the `validate` function
```
with_attr error_message("Invalid nonce") {
      assert tx.signer_nonce = outside_nonce;
}
```
but a check needs to be added to ensure this one match the nonce in Kakarot If I am not mistaken
In the reference implementation there is a `is_valid_outside_execution_nonce` function.